pkg/syncer/cnsoperator/manager/init.go

```diff
@@ -153,6 +155,26 @@ func InitCnsOperator(ctx context.Context, clusterFlavor cnstypes.CnsClusterFlavo
 				}
 			}
 		}()
+	} else if clusterFlavor == cnstypes.CnsClusterFlavorVanilla {
+		// Initialize the k8s orchestrator interface
+		cnsOperator.coCommonInterface, err = commonco.GetContainerOrchestratorInterface(ctx, common.Kubernetes,
```

Comment:
Shouldn't this be protected with the fss check ?

Reply:
I am creating the CO agnostic orchestrator instance here to check for the FSS on line 167. If creation of the CO agnostic orchestrator fails, we will return an error.

Reply:
I think we should move this common code outside of if {} & else if {} blocks:
```cnsOperator.coCommonInterface, err = commonco.GetContainerOrchestratorInterface(ctx, common.Kubernetes, clusterFlavor, coInitParams)
		if err != nil {
			log.Errorf("failed to create CO agnostic interface. Err: %v", err)
			return err
		}
```

Reply:
I thought of this but I will still have to create a if else outside as the parameters take in cluster flavor. Isn't it better to do it here itself?

Reply:
> I thought of this but I will still have to create a if else outside as the parameters take in cluster flavor

You can pass the `clusterFlavor` as a param, right ?
cnsOperator.coCommonInterface, err = commonco.GetContainerOrchestratorInterface(ctx, common.Kubernetes, clusterFlavor, coInitParams)

Reply:
Oops, did not notice that, my bad. Changed it